File: SoftUniCourses/SoftUniBasics/whileloops/ExamPreparation.java
package whileloops;

import java.util.Scanner;

public class ExamPreparation {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int poorGradesLimit =Integer.parseInt(scanner.nextLine());


        String task = scanner.nextLine();
        //четем задачата 1-та,дакато кам. е различн от Enough
        int badGrades = 0;
        int allGradesSum = 0;
        int gradesCount = 0;//бр.н всички задачи
        String currentTask = "";//името на последната задача




        while (!task.equals("Enough")){
            //трябва да четем оценката grade
            int grade = Integer.parseInt(scanner.nextLine());
            allGradesSum += grade;
            gradesCount++;

            if (grade <= 4){
                badGrades++;
                if (badGrades == poorGradesLimit){
                    break;
                }

            }
            currentTask = task;
            task = scanner.nextLine();


        }//kr while
       double averageScore = 1.0 * allGradesSum / gradesCount ;

        // средна оценка
        String output = "";
        if (badGrades ==poorGradesLimit){
            output = String.format("You need a break, %d poor grades.",badGrades);

        } else {
            output = String.format("Average score: %.2f%n" + "Number of problems: %d%n"
            + "Last problem: %s",averageScore,gradesCount,currentTask);
        }

        System.out.println(output);

    }
}

File: SoftUniCourses/JavaScript/VSCode/JavaScript/whileExe/examPreparation.js
function exam(input) {
    let i = 0;
    let failedGrade = Number(input[i]);
    i++;

    let command = input[i];
    i++;

    let badGrades = 0;//бр.нa nezodovolitelni otsenki
    let sumScore = 0;
    let problemCount = 0;//бр.нa всички задачи
    let lastTask = " ";//името на последната задача

    while (command !== "Enough") {
        let task = command;
        let grade = Number(input[i]);
        i++;

        sumScore = sumScore + grade;
        problemCount++;
        lastTask = task;

        if (grade <= 4) {
            badGrades++;
        }

        if (badGrades === failedGrade) {
          console.log(`You need a break, ${badGrades} poor grades.`);
           break;
        }
         command = input[i];
         i++;
      }
      if (command === "Enough") {
        let averageScore = sumScore / problemCount;
        console.log(`Average score: ${averageScore.toFixed(2)}`);
        console.log(`Number of problems: ${problemCount}`);
        console.log(`Last problem: ${lastTask}`);
    }
}
exam(["3", "Money", "6", "Story", "4", "Spring Time", "5", "Bus", "6", "Enough"]);
